File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        var set = new HashSet<int>(nums);
        int longest = 0;

        foreach (var num in set) {
            if (!set.Contains(num - 1)) {
                int len = 1;

                while (set.Contains(num + len)) {
                    len++;
                }

                longest = Math.Max(longest, len);
            }
        }

        return longest;
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    private int bSearch(int[] nums, int target, int l, int r) {
        if (l > r) {
            return -1;
        }

        int m = l + (r - l) / 2;
        if (nums[m] == target) {
            return m;
        } else if (nums[m] < target) {
            return bSearch(nums, target, m + 1, r);
        } else {
            return bSearch(nums, target, l, m - 1);
        }
    }

    public int search(int[] nums, int target) {
        return bSearch(nums, target, 0, nums.length - 1);
    }
}
